esmvaltool/preprocessor/_multimodel.py

```diff
@@ -13,42 +26,330 @@
+    # put in cube
+    stats_cube = iris.cube.Cube(fixed_dspec,
+                                dim_coords_and_dims=cspec,
+                                long_name=stat_name)
+    coord_names = [coord.name() for coord in template_cube.coords()]
+    if 'air_pressure' in coord_names:
+        if len(template_cube.shape) == 3:
+            stats_cube.add_aux_coord(template_cube.
+                                     coord('air_pressure'))
+    stats_cube.attributes['_filename'] = file_name
+    # complete metadata
+    stats_cube.var_name = template_cube.var_name
+    stats_cube.long_name = template_cube.long_name
+    stats_cube.standard_name = template_cube.standard_name
+    stats_cube.units = template_cube.units
     return stats_cube
 
 
-def _sdat(srl_no):
-    """convert to a datatime point"""
-    new_date = dd(1950, 1, 1, 0) + td(srl_no)
+def _to_datetime(delta_t, unit_type):
+    """Convert to a datetime point"""
+    leap_correction = delta_t / (365. * 4.)
+    delta_t = delta_t + leap_correction
+    if unit_type == 'day since 1950-01-01 00:00:00.0000000':
+        new_date = datetime(1950, 1, 1, 0) + timedelta(np.int(delta_t))
+    elif unit_type == 'day since 1850-01-01 00:00:00.0000000':
+        new_date = datetime(1850, 1, 1, 0) + timedelta(np.int(delta_t))
+    # add more supported units here
     return new_date
 
 
 def _get_overlap(cubes):
-    """Get discrete time overlaps."""
-    all_times = [cube.coord('time').points for cube in cubes]
+    """
+    Get discrete time overlaps.
+
+    This method gets the bounds of coord time
+    from the cube and assembles a continuous time
+    axis with smallest unit 1; then it finds the
+    overlaps by doing a 1-dim intersect;
+    takes the floor of first date and
+    ceil of last date.
+    """
+    all_times = []
+    for cube in cubes:
+        # monthly data ONLY
+        # converts all to 365-days calendars
+        time_units = cube.coord('time').units
+        bnd1 = float(cube.coord('time').points[0])
+        bnd2 = float(cube.coord('time').points[-1])
+        if time_units.calendar is not None:
+            if time_units.calendar == '360_day':
+                bnd1 = int(bnd1 / 360.) * 365.
+                bnd2 = (int(bnd2 / 360.) + 1) * 365.
+            else:
+                bnd1 = int(bnd1 / 365.) * 365.
+                bnd2 = (int(bnd2 / 365.) + 1) * 365.
+        else:
+            logger.debug("No calendar type: assuming 365-day")
+            bnd1 = int(bnd1 / 365.) * 365.
+            bnd2 = (int(bnd2 / 365.) + 1) * 365.
+        all_times.append([bnd1, bnd2])
     bounds = [range(int(b[0]), int(b[-1]) + 1) for b in all_times]
-    time_pts = reduce(np.intersect1d, (i for i in bounds))
+    time_pts = reduce(np.intersect1d, bounds)
     if len(time_pts) > 1:
-        return _sdat(float(time_pts[0])), _sdat(float(time_pts[-1]))
+        time_bounds_list = [time_pts[0], time_pts[-1]]
+        return time_bounds_list
 
 
 def _slice_cube(cube, min_t, max_t):
```

Comment:
Is this function still needed?

Reply:
yeah, it's a convenient stock function that spits out a sliced cube on time; is used only for the template cube (so just once), it's very inefficient for many cubes hence the _slice_cube2() function, but the latter one returns only data and not a full cube

Reply:
Ok, maybe you could use cube slicing instead, for more efficiency? See:
http://scitools.org.uk/iris/docs/latest/userguide/subsetting_a_cube.html#cube-indexing

Reply:
@valeriupredoi  Did you see my latest comment here?

Reply:
good call -- slicing the whole cube and not just the data and then we can remove _slice_cube() for good and just keep _slice_cube2(), right? can put this on the to-do when I'll implement the iris datetime handling for the future

Reply:
Yes, that's what I meant.